### PR TITLE
554 refactoring tool all aspects with references

### LINKED ISSUE #554: [Task] Refactoring tool: All Aspects with references to X
Assume that I want to improve a basic model which is referred to (directly or indirectly) by multiple models.
The improvement is so important that I want to create new versions of all models that (directly or indirectly) refer to it.

So with the SDK I want to get 
- a list of all such models (e.g. from a github repository)
- if possible for each file a reference structure that explains to me how it indirectly refers to it, so that I can check and understand it
- if possible an additional option to create new versions of each file automatically (and a parameter that lets me decide if I want to count up the major, minor or bugfix version of it)

### PR DESCRIPTION
**Pull request by Yauhenikapl**
## Description

Please include a summary of the changes and the related issue. List any dependencies that are required for this change.

Fixes #554 

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: tools/samm-cli/pom.xml [hunk 1]
```diff
@@ -129,6 +129,12 @@
          <artifactId>tika-core</artifactId>
          <scope>test</scope>
       </dependency>
+      <dependency>
+         <groupId>org.eclipse.esmf</groupId>
+         <artifactId>esmf-sdk-github-resolver</artifactId>
+         <version>DEV-SNAPSHOT</version>
```
Comment (atextor) [inline tools/samm-cli/pom.xml:135]: Remove `<version>`

File: tools/samm-cli/pom.xml [hunk 2]
```diff
@@ -129,6 +129,12 @@
...
+         <scope>compile</scope>
```
Comment (atextor) [inline tools/samm-cli/pom.xml:136]: Is this right? Doesn't this dependency needs default scope?

File: tools/samm-cli/src/main/java/org/eclipse/esmf/AspectSearchCommand.java [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+import org.eclipse.esmf.aspectmodel.urn.AspectModelUrn;
+import org.eclipse.esmf.metamodel.AspectModel;
+import org.eclipse.esmf.metamodel.ModelElement;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import picocli.CommandLine;
+
+@CommandLine.Command( name = AspectSearchCommand.COMMAND_NAME,
+      description = "Search separate Properties files used by Aspect Model",
```
Comment (atextor) [inline tools/samm-cli/src/main/java/org/eclipse/esmf/AspectSearchCommand.java:35]: I don't think we search Properties files (*.properties). Maybe: "Shows where model elements are used in Aspect Models" or similar?
Maybe "search" is also not optimal as a command name, IMO it is not really a search function, maybe we can call it "usage". Or something else?

File: tools/samm-cli/src/main/java/org/eclipse/esmf/AspectSearchCommand.java [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+   @CommandLine.Mixin
+   private ExternalResolverMixin customResolver;
+
+   @CommandLine.Parameters( paramLabel = "INPUT", description = "Input file name of the Aspect Model .ttl file", arity = "1", index = "0" )
+   private String input;
+
+   @CommandLine.Option( names = { "--models-path", "-msp" }, description = "Path to Models, where have to search" )
+   private String pathToModels = "-";
+
+   @CommandLine.Option( names = { "--github", "-gh" }, description = "Use this flag to introduce what has to be scanned" )
```
Comment (atextor) [inline tools/samm-cli/src/main/java/org/eclipse/esmf/AspectSearchCommand.java:59]:
```suggestion
   @CommandLine.Option( names = { "--github", "-gh" }, description = "Enable loading Aspect Models from GitHub" )
```

File: tools/samm-cli/src/main/java/org/eclipse/esmf/AspectSearchCommand.java [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+   @CommandLine.Mixin
+   private LoggingMixin loggingMixin;
+
+   @CommandLine.Mixin
+   private ExternalResolverMixin customResolver;
+
+   @CommandLine.Parameters( paramLabel = "INPUT", description = "Input file name of the Aspect Model .ttl file", arity = "1", index = "0" )
+   private String input;
+
+   @CommandLine.Option( names = { "--models-path", "-msp" }, description = "Path to Models, where have to search" )
```
Comment (atextor) [inline tools/samm-cli/src/main/java/org/eclipse/esmf/AspectSearchCommand.java:56]:
```suggestion
   @CommandLine.Option( names = { "--models-root", "-mr" }, description = "Set the models root directory" )
```

File: tools/samm-cli/src/main/java/org/eclipse/esmf/AspectSearchCommand.java [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+   private String pathToModels = "-";
+
+   @CommandLine.Option( names = { "--github", "-gh" }, description = "Use this flag to introduce what has to be scanned" )
+   private boolean searchInGithub = false;
+
+   @CommandLine.Option( names = { "--github-name", "-ghn" }, description = "Provide GitHub name. Example: eclipse-esmf/esmf-sdk." )
```
Comment (atextor) [inline tools/samm-cli/src/main/java/org/eclipse/esmf/AspectSearchCommand.java:62]:
```suggestion
   @CommandLine.Option( names = { "--github-name", "-ghn" }, description = "Set the GitHub repository name. Example: eclipse-esmf/esmf-sdk." )
```

File: tools/samm-cli/src/main/java/org/eclipse/esmf/AspectSearchCommand.java [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+   private String gitHubName = "";
+
+   @CommandLine.Option( names = { "--github-directory", "-ghd" }, description = "Provide GitHub directory for scanning" )
```
Comment (atextor) [inline tools/samm-cli/src/main/java/org/eclipse/esmf/AspectSearchCommand.java:65]:
```suggestion
   @CommandLine.Option( names = { "--github-directory", "-ghd" }, description = "Set the GitHub directory" )
```

File: tools/samm-cli/src/main/java/org/eclipse/esmf/AspectSearchCommand.java [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+   private String gitHubDirectory = "";
+
+   @CommandLine.Option( names = { "--github-branch", "-ghb" }, description = "Provide GitHub branch for scanning" )
```
Comment (atextor) [inline tools/samm-cli/src/main/java/org/eclipse/esmf/AspectSearchCommand.java:68]:
```suggestion
   @CommandLine.Option( names = { "--github-branch", "-ghb" }, description = "Set the GitHub branch" )
```

**Review by atextor — CHANGES_REQUESTED** [review]

File: tools/samm-cli/src/main/java/org/eclipse/esmf/AspectSearchCommand.java (hunk 3 above)
Comment (Yauhenikapl) [reply tools/samm-cli/src/main/java/org/eclipse/esmf/AspectSearchCommand.java:35]: Yes, here you are totally right, 100% we have to change it for better explanation.

I think, that 'usage' as command name and 'Shows where model elements are used in Aspect Models' like description would be great!

**Review by atextor — APPROVED** [review]

**Pull request merged**